src/main/java/com/mongodb/jdbc/mongosql/TranslateResult.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2024-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.jdbc.mongosql;
+
+import com.mongodb.jdbc.JsonSchema;
+import com.mongodb.jdbc.MongoJsonSchema;
+import java.util.List;
+import org.bson.BsonDocument;
+import org.bson.codecs.pojo.annotations.BsonCreator;
+import org.bson.codecs.pojo.annotations.BsonProperty;
+
+public class TranslateResult {
+    public String targetDb;
+    public String targetCollection;
+    public List<BsonDocument> pipeline;
+    public MongoJsonSchema resultSetSchema;
+    public List<List<String>> selectOrder;
+    public String error;
+    public Boolean errorIsInternal;
+
+    @BsonCreator
+    public TranslateResult(
```

Comment:
I added the BaseResult class and moved the error handling to runCommand as suggested.  

Somehow this caused an issue with finding the codec for `TranslateResult` because of the difference in the `JsonSchema` to `MongoJsonSchema` types so I moved that conversion to the `executeDirectClusterQuery()` function.

Reply:
uhm, odd. Did you try to do something similar to what is done in MongoVersionedJsonSchema?
Don't annotate resultSetSchema with `@BsonProperty("result_set_schema")` and in your constructor do `this.resultSetSchema = MongoJsonSchema.toSimplifiedMongoJsonSchema(resultSetSchema)`. I believe this should work.